### ISSUE TITLE
Typescript: trailing comma in interface is not allowed

### ISSUE THREAD

**Issue by dotlouis** [prettier/prettier#1858]
Hi, I just ran prettier on my code base and had only a single issue which comes from an interface.

[Demo](https://prettier.github.io/prettier/#%7B%22content%22%3A%22import%20%7B%20Menu%20%7D%20from%20'.%2Fmenu'%3B%5Cn%5Cnexport%20interface%20ShopQueryResult%20%7B%5Cn%20%20chic%3A%20boolean%3B%5Cn%20%20location%3A%20number%5B%5D%3B%5Cn%20%20menus%3A%20Menu%5B%5D%3B%5Cn%20%20openingDays%3A%20number%5B%5D%3B%5Cn%20%20closingDays%3A%20%5B%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20from%3A%20string%2C%5Cn%20%20%20%20%20%20to%3A%20string%2C%5Cn%20%20%20%20%7D%5Cn%20%20%5D%3B%5Cn%20%20shop%3A%20string%3B%5Cn%20%20distance%3A%20number%3B%5Cn%7D%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22es5%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D)

generated TS:
```TS
export interface ShopQueryResult {
  chic: boolean;
  location: number[];
  menus: Menu[];
  openingDays: number[];
  closingDays: [
    {
      from: string,
      to: string,
    }, // <== this one
  ];
  shop: string;
  distance: number;
}
```

It basically comes down to the trailing comma inside the `closingDays` array, after the object literal.
Typescript compiler is not happy with it. Don't know why. My guess is that it should be accepted like any other trailing commas but it's probably easier to fix prettier than typescript.

Thanks a lot for this awesome tool, I look forward to use it a lot more,
Cheers

**Comment by vjeux** [comment]
cc @DanielRosenwasser, another place where typescript doesn't support trailing commas

**Comment by DanielRosenwasser** [comment]
The problem is not the interface, the problem is that tuple types are not allowed to have trailing commas. 

Something to keep in mind: did you actually mean to write a tuple type, or did you mean to write `{from: string, to: string}[]` or `Array<{from: string, to: string}>`?

**Comment by dotlouis** [comment]
You are absolutely right, that's what I meant !
Thanks for the prompt answer !

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!

**Comment by DanielRosenwasser** [comment]
Great! For the record, this is probably a good argument against trailing commas in tuple types - while I think it makes sense for object literals and argument lists, I find it more questionable in this context. 😃
